monkeytype/typing.py

```diff
@@ -26,13 +27,55 @@
 
 
 DUMMY_TYPED_DICT_NAME = 'DUMMY_NAME'
+DUMMY_REQUIRED_TYPED_DICT_NAME = 'REQUIRED_TYPED_DICT_NAME'
+DUMMY_OPTIONAL_TYPED_DICT_NAME = 'OPTIONAL_TYPED_DICT_NAME'
 
 
 # Functions like shrink_types and get_type construct new types at runtime.
 # Mypy cannot currently type these functions, so the type signatures for this
 # file live in typing.pyi.
 
 
+def types_equal(typ, other_type) -> bool:
+    # Types for which equality with inner TypedDict doesn't work correctly on 3.6.
```

Comment:
The comment here suggests this is only necessary for 3.6, but it seems like we use this function unconditionally on any Python version? The reason I suggested putting this in `compat.py` is that I thought we could isolate the hackiness to 3.6 and maybe use normal equality checks on 3.7?

Reply:
Yeah, that's what I was going for, but got stymied by the import error.

Turns out that all imports in compat.py had the error and had needed `type: ignore` :)

Reply:
The best solution would be to override `List.__eq__`, just like we do for `TypedDict`, but that didn't work in the REPL when I tried it. I'm going to leave it as is for now, since we use it in only one place.